app/views/collections/show.rb

```diff
@@ -172,25 +172,26 @@ def stylised_collection_stats
 
     # @todo refactor to:
     # - make modular have this be a concern
-    # - lookup proper search title & type using labeling from  presenters/concerns/facet/labelling.rb
-    # - refactor facet_entry_field_title to not be called for each facet entry
     def preview_search_data
       return nil if @landing_page.facet_entries.blank?
 
       @landing_page.facet_entries.map do |facet_entry|
+        presenter = facet_entry_presenter(facet_entry)
         {
-          preview_search_title: facet_entry.title,
-          preview_search_type: facet_entry_field_title(facet_entry),
+          preview_search_title: presenter.facet_item_label(facet_entry.facet_value) || facet_entry.title,
+          preview_search_type: presenter.facet_title || facet_entry.facet_field,
           preview_search_url: browse_entry_url(facet_entry, @landing_page, format: 'json'),
           preview_search_more_link: browse_entry_url(facet_entry, @landing_page)
         }
       end
     end
 
-    def facet_entry_field_title(facet_entry)
-      ff = Europeana::Blacklight::Response::Facets::FacetField.new(facet_entry.facet_field, [])
-      presenter = FacetPresenter.build(ff, controller)
-      presenter.facet_title || facet_entry.facet_field
+    def facet_entry_presenter(facet_entry)
+      @facet_presenters ||= {}
```

Comment:
If this method is only ever called by `preview_search_data` where the presenter is assigned to a variable, then I'm not sure this object caching gains us anything?